include/deal.II/grid/tria.h

```diff
@@ -4181,6 +4356,88 @@ namespace internal
 } // namespace internal
 
 
+
+namespace construction
+{
+  template <int dim>
+  template <class Archive>
+  void
+  CellData<dim>::serialize(Archive &ar, const unsigned int /*version*/)
+  {
+    ar &id;
+    ar &subdomain_id;
+    ar &level_subdomain_id;
+    ar &manifold_id;
+    if (dim >= 2)
+      ar &manifold_line_ids;
+    if (dim >= 3)
+      ar &manifold_quad_ids;
+    ar &boundary_ids;
+  }
+
+
+  template <int dim, int spacedim>
+  template <class Archive>
+  void
+  ConstructionData<dim, spacedim>::serialize(Archive &ar,
+                                             const unsigned int /*version*/)
+  {
+    ar &coarse_cells;
+    ar &coarse_cell_vertices;
+    ar &coarse_cell_index_to_coarse_cell_id;
+    ar &cell_infos;
+    ar &settings;
+  }
+
+
+
+  template <int dim>
+  bool
+  CellData<dim>::operator==(const CellData<dim> &other) const
+  {
+    if (this->id != other.id)
+      return false;
+    if (this->subdomain_id != other.subdomain_id)
+      return false;
```

Comment:
You can simplify this and the next function by just saying
```
  return ((this->id == other.id)
             &&
            (this->subdomain_id == other.subdomain_id)
            &&
            ...);
```
The compiler internally converts that into exactly what you wrote, i.e., the chain of evaluations is terminated once it encounters the first `false` comparison (because then the rest of the comparisons no longer matter).

Reply:
I leave this for the future.